Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -1325,42 +1325,80 @@ extension Tensor.IndexPath {
 //===------------------------------------------------------------------------------------------===//
 
 extension Tensor {
-  /// Returns `true` if the given axis is in the range `[-rank, rank)`.
+  /// Returns `true` iff `k` denotes an axis of `self`.
   @usableFromInline
-  internal func isAxisInRange<T: BinaryInteger>(_ axis: T) -> Bool {
-    let axis = Int(axis)
+  internal func isValid<T: BinaryInteger>(axis k: T) -> Bool {
+    let axis = Int(k)
     return axis >= -rank && axis < rank
   }
 
-  /// Returns `true` if the given scalar tensor is in the range `[-rank, rank)`.
+  /// Returns `true` iff each element of `dims` denotes an axis of `self`.
```

Comment:
IMO `dims` is a bit worse than something meaningless like `r`

1. It is an abbrev (strike one)
2. it has a meaning when not read as an abbrev., which makes it confusing.  This has nothing to do with dimming
3. When you do interpret it correctly,  it doesn't seem apt.  This isn't describing the dimensions of something, which is what you would expect from a parameter called “dimensions”

Actually I'm only assuming you mean “dimensions”, because it's an abbrev.  ;-)

Reply:
I know you're going to chafe every time I refer to [this](https://swift.org/documentation/api-design-guidelines/#choose-parameter-names-to-serve-doc), but it doesn't matter whether the name helps readability in the body of your function; you can always create a new binding there.

Reply:
Oh, I have no objections to [that](https://swift.org/documentation/api-design-guidelines/#choose-parameter-names-to-serve-doc); I've reverted to just `axes`.